controllers/clusterresources/redisuser_controller.go

```diff
@@ -134,7 +136,7 @@ func (r *RedisUserReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	for clusterID, event := range user.Status.ClustersEvents {
 		if event == models.CreatingEvent {
 			_, err = r.API.CreateRedisUser(user.Spec.ToInstAPI(password, clusterID, username))
-			if err != nil {
+			if err != nil && !strings.Contains(fmt.Sprint(err), "username already in use") {
```

Comment:
`!strings.Contains(fmt.Sprint(err)` - doesn't seem like the best approach for error handling. If you don't get the HTTP status code `AlreadyExist`, I recommend implementing the following alternative strategy:
1. Retrieve a list of users from the Instaclustr API.
2. Iterate through this list and check for any duplicate users.
3. Only proceed with sending a request for user creation if no duplicates are found in the user list.

Reply:
fixed